### ISSUE TITLE
Support Faraday 1.0 and above

### ISSUE THREAD

**Issue by dlpierce** [samvera/active_fedora#1479]
```
Bundler could not find compatible versions for gem "faraday":
  In Gemfile:
    hyrax (~> 2.9.6) was resolved to 2.9.6, which depends on
      active-fedora (>= 11.5.2, < 13) was resolved to 12.2.3, which depends on
        faraday (~> 0.12)

    oai (~> 1.2) was resolved to 1.2.1, which depends on
      faraday-follow_redirects (< 2, >= 0.3.0) was resolved to 0.3.0, which depends on
        faraday (< 3, >= 1)
```

**Comment by jrgriffiniii** [comment]
4b0e79acca00c31057c1f3fe2023df0cf58845fb  raises only 5 errors now:

```bash
Finished in 1 minute 44.16 seconds (files took 3.16 seconds to load)
23 examples, 5 failures

Failed examples:

rspec ./spec/integration/relation_spec.rb:27 # ActiveFedora::Base is expected to be empty
rspec ./spec/integration/relation_spec.rb:28 # ActiveFedora::Base is expected not to be present
rspec ./spec/integration/relation_spec.rb:25 # ActiveFedora::Base is expected to eq false
rspec ./spec/integration/relation_spec.rb:26 # ActiveFedora::Base is expected to be blank
rspec ./spec/integration/relation_spec.rb:72 # ActiveFedora::Base when some records exist #each when called from Base yields al
l items
```

**Comment by jrgriffiniii** [comment]
https://github.com/samvera/active_fedora/compare/main...issues-1479-jrgriffiniii-faraday1.0 has progressed to the point where `ActiveFedora::FileIO` is refactored, and the tests are passing with an updated RuboCop configuration.

**Comment by dlpierce** [comment]
This appears to have been resolved as part of #1485, which still needs to be released.
